sycl/include/CL/sycl/detail/clusm.hpp

```diff
@@ -108,23 +105,24 @@ class CLUSM {
 } // namespace usm
 
 namespace cliext {
-bool initializeExtensions(cl_platform_id platform);
+bool initializeExtensions(cl_context context, cl_platform_id platform);
 } // namespace cliext
 
 } // namespace detail
 } // namespace sycl
 } // namespace cl
 
-__SYCL_EXPORTED extern cl::sycl::detail::usm::CLUSM *gCLUSM;
-inline cl::sycl::detail::usm::CLUSM *GetCLUSM() {
-  if (gCLUSM == nullptr) {
-    cl::sycl::detail::usm::CLUSM::Create(gCLUSM);
-  }
-
+__SYCL_EXPORTED extern std::map<cl_context, cl::sycl::detail::usm::CLUSM *>
```

Comment:
Can't you store pointer to CLUSM in the context object?

Reply:
Yep.  Wasn't sure if you guys preferred it to be that invasive.

Reply:
Actually - not sure I can really do this.

Reply:
See below in the opencl_shim part.